Automattic/jetpack
# Story block: Handle blavatar_url for site icon

Fixes https://github.com/Automattic/story-block/issues/13.

#### Changes proposed in this Pull Request:
It seems that for some sites `get_site_icon_url` returns the fallback value, even when a custom site icon has been set. In those cases `blavatar_url` needs to be used. I'm not sure what the conditions for this are, I haven't been able to reproduce it on my own sites, but I have an example site where this occurs I can share.

I noticed that the `Jetpack_PostImages` class (and a few other places) has [logic for using `blavatar_` functions in addition to `get_site_icon_url`](https://github.com/Automattic/jetpack/blob/ba2dd6c30be137baad96aafb2466d0aed7e182e5/projects/plugins/jetpack/class.jetpack-post-images.php#L479-L494), so I added similar functionality to the story block. That seems to resolve the issue.

#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions:

* On a site with a custom site icon, play a story and make sure the site icon shows up at the top
* For a site that relies on blavatar (ping me for an example), play a story and make sure the site icon shows up
